### PR TITLE
Create ClassLoaders inside doPrivileged() + Auto-close the WAR file after exploding it

### PR DESCRIPTION
**Pull request by recrsn**
Partially addresses #374

### CODE REVIEW

File: bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java [hunk 1]
```diff
@@ -338,12 +341,12 @@ public int run() throws Throwable {
         }
     }
 
-    public ClassLoader createJenkinsWarClassLoader() throws IOException, NoSuchMethodException, InvocationTargetException, IllegalAccessException {
```
Comment (oleg-nenashev) [inline bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java:341]: I do not recall what led to this exception list

File: bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java [hunk 2]
```diff
@@ -338,12 +341,12 @@ public int run() throws Throwable {
...
-        return new ClassLoaderBuilder(new SideClassLoader(getPlatformClassloader()))
-                .collectJars(new File(warDir,"WEB-INF/lib"))
+    public ClassLoader createJenkinsWarClassLoader() throws PrivilegedActionException {
+        return AccessController.doPrivileged((PrivilegedExceptionAction<ClassLoader>) () -> new ClassLoaderBuilder(new SideClassLoader(getPlatformClassloader()))
```
Comment (oleg-nenashev) [inline bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java:345]: Jenkins is unlikely to work on advanced security policies . Should be fine though I will need to add Autotests for the resource loading

**Review by oleg-nenashev — APPROVED** [review]
Should be good though I will need to add a test for the AccessController and WAR resource loading

**Review by oleg-nenashev — APPROVED** [review]
I tested it locally to verify the workaround for https://github.com/jenkinsci/jenkinsfile-runner/issues/413, and I confirm it still works well after the patch

File: bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java (hunk 2 above)
Comment (jglick) [reply bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java:345]: This is pointless. See https://github.com/spotbugs/spotbugs/issues/1515

**Pull request merged**